models/maverick/base/maverick_v1_base_trades.sql

```diff
@@ -0,0 +1,106 @@
+    file_format = 'delta',
+    incremental_strategy = 'merge',
+    unique_key = ['block_date', 'blockchain', 'project', 'version', 'tx_hash', 'evt_index'],
+    post_hook='{{ expose_spells(\'["base"]\',
+                                "project",
+                                "maverick_v1",
+                                \'["get620v"]\') }}'
+    )
+}}
+
+{% set project_start_date = '2023-01-01' %}
+
+WITH dexs AS
+(
+    --Maverick V1
+    SELECT
+        t.evt_block_time AS block_time
+        ,t.recipient AS taker
+        ,CAST(NULL AS VARBINARY) AS maker
+        ,amountOut AS token_bought_amount_raw
+        ,amountIn AS token_sold_amount_raw
+        ,NULL AS amount_usd
+        ,CASE WHEN tokenAIn THEN f.tokenB ELSE f.tokenA END AS token_bought_address
+        ,CASE WHEN tokenAIn THEN f.tokenA ELSE f.tokenB END AS token_sold_address
+        ,t.contract_address as project_contract_address
+        ,t.evt_tx_hash AS tx_hash
+        ,t.evt_index
+    FROM
+        {{ source('maverick_v1_base', 'pool_evt_Swap') }} t
+    INNER JOIN {{ source('maverick_v1_base', 'factory_evt_PoolCreated') }} f
+        ON f.poolAddress = t.contract_address
+    {% if is_incremental() %}
+    WHERE t.evt_block_time >= date_trunc('day', now() - interval '7' day)
+    {% endif %}
+)
+SELECT
+    'base' AS blockchain
+    ,'maverick' AS project
+    ,'1' AS version
+    ,TRY_CAST(date_trunc('day', dexs.block_time) AS date) AS block_date
+    ,CAST(date_trunc('month', dexs.block_time) AS date) AS block_month
+    ,dexs.block_time
+    ,erc20a.symbol AS token_bought_symbol
+    ,erc20b.symbol AS token_sold_symbol
+    ,case
+        when lower(erc20a.symbol) > lower(erc20b.symbol) then concat(erc20b.symbol, '-', erc20a.symbol)
+        else concat(erc20a.symbol, '-', erc20b.symbol)
+    end as token_pair
+    ,dexs.token_bought_amount_raw / power(10, erc20a.decimals) AS token_bought_amount
+    ,dexs.token_sold_amount_raw / power(10, erc20b.decimals) AS token_sold_amount
+    ,dexs.token_bought_amount_raw  AS token_bought_amount_raw
+    ,dexs.token_sold_amount_raw  AS token_sold_amount_raw
+    ,coalesce(
+        dexs.amount_usd
+        ,(dexs.token_bought_amount_raw / power(10, p_bought.decimals)) * p_bought.price
+        ,(dexs.token_sold_amount_raw / power(10, p_sold.decimals)) * p_sold.price
+    ) AS amount_usd
+    ,dexs.token_bought_address
+    ,dexs.token_sold_address
+    ,coalesce(dexs.taker, tx."from") AS taker -- subqueries rely on this COALESCE to avoid redundant joins with the transactions table
+    ,dexs.maker
+    ,dexs.project_contract_address
+    ,dexs.tx_hash
+    ,tx."from" AS tx_from
+    ,tx.to AS tx_to
+    ,dexs.evt_index
+FROM dexs
+INNER JOIN {{ source('base', 'transactions') }} tx
+    ON tx.hash = dexs.tx_hash
+    {% if not is_incremental() %}
+    AND tx.block_time >= TIMESTAMP '{{project_start_date}}'
+    {% endif %}
+    {% if is_incremental() %}
+    AND tx.block_time >= date_trunc('day', now() - interval '7' day)
```

Comment:
We can use `incremental_predicate` to rewrite all incremental filters:
```suggestion
    AND {{ incremental_predicate('tx.block_time') }}
```